publisher/publisher.go

```diff
@@ -38,14 +38,23 @@ type logCache struct {
 // AddLog adds a *Log to the cache by constructing the statName, client and
 // verifier for the given uri & base64 public key.
 func (c *logCache) AddLog(uri, b64PK string) (*Log, error) {
-	c.Lock()
-	defer c.Unlock()
-
+	// Lock the mutex for reading to check the cache
+	c.RLock()
 	// If we have already added this log, give it back
 	if log, present := c.logs[b64PK]; present {
+		// Make sure to RUnlock() after return
+		defer c.RUnlock()
```

Comment:
Great feedback, thanks! Implemented in https://github.com/letsencrypt/boulder/pull/2342/commits/42ec8c259b6d1b211143cf017801e76b50e07666